src/Arriba/Arriba.Composition/Composition/Host.cs

```diff
@@ -55,6 +56,26 @@ public static void AddArribaServices(this IServiceCollection services, ISecurity
             services.AddSingleton<ApplicationServer, ComposedApplicationServer>();
         }
 
+        private static void AddSecureDatabase(this IServiceCollection services)
+        {
+            var sp = services.BuildServiceProvider();
```

Comment:
Using BuildServiceProvider() is a red flag to me. There are cases where this should be used, however I don't think this is one of them. I might suggest rewriting this as follows.
```suggestion
            services.AddSingleton<SecureDatabase>(p => {
                 var databaseFactory = p.GetService<DatabaseFactory>();

                var secureDatabase = databaseFactory.GetDatabase();

            //HACK: As the crawler don't have the AppRoles available at the JWT Token as claims
            //needed to grant for each role (PermissionScope) the roles name must use the same name as the Scope.
            secureDatabase.TableNames.ToList().ForEach(table =>
            {
                var security = secureDatabase.Security(table);
                security.Grant(IdentityScope.Group, PermissionScope.Owner.ToString(), PermissionScope.Owner);
                security.Grant(IdentityScope.Group, PermissionScope.Writer.ToString(), PermissionScope.Writer);
                security.Grant(IdentityScope.Group, PermissionScope.Reader.ToString(), PermissionScope.Reader);
            });

            return secureDatabase;
          });
```